static/dependencies.config.php

```diff
@@ -47,6 +47,13 @@
 use Friendica\Util;
 use Psr\Log\LoggerInterface;
 
+/**
+ * @var string $basepath The base path of the Friendica installation without trailing slash
+ */
+$basepath = (function() {
+	return dirname(__FILE__, 2);
```

Comment:
And this is always true? Why do we even have the `BasePath` class if this is the case?

And is it a good idea to include an execution inside a dependency file? It doesn't sound right - what about mocking the BasePath and loading the dependency file from a virtual drive because of the unmockable `__FILE__` constant, that wouldn't be possible at least afaik, would it?

Reply:
> And this is always true? Why do we even have the `BasePath` class if this is the case?

That's the question I worked on yesterday. I could not find any place where `dirname()` of `__FILE__` would not return a valid dir path. Thats why I could condense the `BasePath::getPath()` to this line of code.

> And is it a good idea to include an execution inside a dependency file? It doesn't sound right -

I just copied the call from line 67, see rules for `'$basepath'`.

I would prefer to move the call inside a new `bootstrap.php` file and move the `dependencies.config.php` file into a config folder.

> what about mocking the BasePath and loading the dependency file from a virtual drive because of the unmockable `__FILE__` constant, that wouldn't be possible at least afaik, would it?

I'm not sure if I understand you right. Can you show me a use case where this would not work? I would like to add it to the `BasePath` test file.